Hymns/Common/SearchBar.swift

```diff
@@ -1,67 +1,78 @@
 import SwiftUI
 
-struct SearchBar: UIViewRepresentable {
+/**
+ * Custom search bar that will animate in a `Cancel` button when activated/selected.
+ * https://stackoverflow.com/questions/56490963/how-to-display-a-search-bar-with-swiftui
+ */
+struct SearchBar: View {
 
-    @Binding var text: String
-    var selectedOnAppear: Bool = false //when true will make a search bar first responder
+    @Binding var searchText: String
+    @Binding var searchActive: Bool
+    let placeholderText: String
 
-    class Coordinator: NSObject, UISearchBarDelegate {
+    var body: some View {
+        HStack {
+            HStack {
+                Image(systemName: "magnifyingglass")
+                TextField(placeholderText, text: $searchText, onEditingChanged: { _ in
+                    withAnimation(.easeInOut(duration: 0.3)) {
+                        self.searchActive = true
+                    }
+                }, onCommit: {
+                    print("onCommit")
+                }).foregroundColor(.primary)
 
-        @Binding var text: String
-        var didBecomeFirstResponder = false
+                Button(action: {self.searchText = ""}, label: {Image(systemName: "xmark.circle.fill").opacity(self.searchText.isEmpty ? 0.0 : 1.0)})
+            }
+            .padding(EdgeInsets(top: 8, leading: 6, bottom: 8, trailing: 6))
+            .foregroundColor(.secondary)
+            .background(Color(.secondarySystemBackground))
+            .cornerRadius(CGFloat(integerLiteral: 10))
 
-        init(text: Binding<String>) {
-            _text = text
+            if searchActive {
+                Button("Cancel") {
+                    // this must be placed before the other commands here
+                    UIApplication.shared.endEditing(true)
+                    self.searchText = ""
+                    withAnimation {
+                        self.searchActive = false
+                    }
+                }
+                .foregroundColor(Color(.systemBlue))
+                .transition(AnyTransition.move(edge: .trailing))
+                .animation(.easeInOut(duration: 0.3))
```

Comment:
.3 feels a bit slow especially considering a user is ready to quickly look up a hymn. Looks great though as far as the animation. I tested .1 and .2 and to me it seems like .2 is an optimal amount of animation time. Up to you though, just something to consider